File: Android/sources/com/google/android/gms/common/api/GoogleApiActivity.java
package com.google.android.gms.common.api;

import android.app.Activity;
import android.app.PendingIntent;
import android.content.ActivityNotFoundException;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.IntentSender;
import android.os.Build;
import android.os.Bundle;
import com.google.android.gms.common.annotation.KeepName;

@KeepName
public class GoogleApiActivity extends Activity implements DialogInterface.OnCancelListener {
    public static final /* synthetic */ int b = 0;
    public int a = 0;

    public final void onActivityResult(int i, int i2, Intent intent) {
        super.onActivityResult(i, i2, intent);
        if (i == 1) {
            boolean booleanExtra = getIntent().getBooleanExtra("notify_manager", true);
            this.a = 0;
            setResult(i2, intent);
            if (booleanExtra) {
                ht6 f = ht6.f(this);
                if (i2 == -1) {
                    ws9 ws9 = f.v0;
                    ws9.sendMessage(ws9.obtainMessage(3));
                } else if (i2 == 0) {
                    f.g(new bi3(13, (PendingIntent) null), getIntent().getIntExtra("failing_client_id", -1));
                }
            }
        } else if (i == 2) {
            this.a = 0;
            setResult(i2, intent);
        }
        finish();
    }

    public final void onCancel(DialogInterface dialogInterface) {
        this.a = 0;
        setResult(0);
        finish();
    }

    public final void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        if (bundle != null) {
            this.a = bundle.getInt("resolution");
        }
        if (this.a != 1) {
            Bundle extras = getIntent().getExtras();
            if (extras == null) {
                finish();
                return;
            }
            PendingIntent pendingIntent = (PendingIntent) extras.get("pending_intent");
            Integer num = (Integer) extras.get("error_code");
            if (pendingIntent == null && num == null) {
                finish();
            } else if (pendingIntent != null) {
                try {
                    startIntentSenderForResult(pendingIntent.getIntentSender(), 1, (Intent) null, 0, 0, 0);
                    this.a = 1;
                } catch (ActivityNotFoundException unused) {
                    if (extras.getBoolean("notify_manager", true)) {
                        ht6.f(this).g(new bi3(22, (PendingIntent) null), getIntent().getIntExtra("failing_client_id", -1));
                    } else {
                        String k = wj6.k("Activity not found while launching ", pendingIntent.toString(), ".");
                        if (Build.FINGERPRINT.contains("generic")) {
                            k.concat(" This may occur when resolving Google Play services connection issues on emulators with Google APIs but not Google Play Store.");
                        }
                    }
                    this.a = 1;
                    finish();
                } catch (IntentSender.SendIntentException unused2) {
                    finish();
                }
            } else {
                vzg.m(num);
                dt6.d.c(this, num.intValue(), 2, this);
                this.a = 1;
            }
        }
    }

    public final void onSaveInstanceState(Bundle bundle) {
        bundle.putInt("resolution", this.a);
        super.onSaveInstanceState(bundle);
    }
}

File: IOS/v2/com/google/android/gms/common/api/GoogleApiActivity.swift
package com.google.iOS.gms.common.api;.swift
import iOS.app.Activity;.swift
import iOS.app.PendingIntent;.swift
import iOS.content.ActivityNotFoundException;.swift
import iOS.content.DialogInterface;.swift
import iOS.content.Intent;.swift
import iOS.content.IntentSender;.swift
import iOS.os.Build;.swift
import iOS.os.Bundle;.swift
import com.google.iOS.gms.common.annotation.KeepName;.swift
@KeepName.swift
public class GoogleApiActivity extends Activity implements DialogInterface.OnCancelListener {.swift
public static final /* synthetic */ int b = 0;.swift
public int a = 0;.swift
public final void onActivityResult(int i, int i2, Intent intent) {.swift
super.onActivityResult(i, i2, intent);.swift
if (i == 1) {.swift
boolean booleanExtra = getIntent().getBooleanExtra("notify_manager", true);.swift
this.a = 0;.swift
setResult(i2, intent);.swift
if (booleanExtra) {.swift
ht6 f = ht6.f(this);.swift
if (i2 == -1) {.swift
ws9 ws9 = f.v0;.swift
ws9.sendMessage(ws9.obtainMessage(3));.swift
} else if (i2 == 0) {.swift
f.g(new bi3(13, (PendingIntent) null), getIntent().getIntExtra("failing_client_id", -1));.swift
}.swift
}.swift
} else if (i == 2) {.swift
this.a = 0;.swift
setResult(i2, intent);.swift
}.swift
finish();.swift
}.swift
public final void onCancel(DialogInterface dialogInterface) {.swift
this.a = 0;.swift
setResult(0);.swift
finish();.swift
}.swift
public final void onCreate(Bundle bundle) {.swift
super.onCreate(bundle);.swift
if (bundle = null) {.swift
this.a = bundle.getInt("resolution");.swift
}.swift
if (this.a = 1) {.swift
Bundle extras = getIntent().getExtras();.swift
if (extras == null) {.swift
finish();.swift
return;.swift
}.swift
PendingIntent pendingIntent = (PendingIntent) extras.get("pending_intent");.swift
Integer num = (Integer) extras.get("error_code");.swift
if (pendingIntent == null && num == null) {.swift
finish();.swift
} else if (pendingIntent = null) {.swift
try {.swift
startIntentSenderForResult(pendingIntent.getIntentSender(), 1, (Intent) null, 0, 0, 0);.swift
this.a = 1;.swift
} catch (ActivityNotFoundException unused) {.swift
if (extras.getBoolean("notify_manager", true)) {.swift
ht6.f(this).g(new bi3(22, (PendingIntent) null), getIntent().getIntExtra("failing_client_id", -1));.swift
} else {.swift
String k = wj6.k("Activity not found while launching ", pendingIntent.toString(), ".");.swift
if (Build.FINGERPRINT.contains("generic")) {.swift
k.concat(" This may occur when resolving Google Play services connection issues on emulators with Google APIs but not Google Play Store.");.swift
}.swift
}.swift
this.a = 1;.swift
finish();.swift
} catch (IntentSender.SendIntentException unused2) {.swift
finish();.swift
}.swift
} else {.swift
vzg.m(num);.swift
dt6.d.c(this, num.intValue(), 2, this);.swift
this.a = 1;.swift
}.swift
}.swift
}.swift
public final void onSaveInstanceState(Bundle bundle) {.swift
bundle.putInt("resolution", this.a);.swift
super.onSaveInstanceState(bundle);.swift
}.swift
}.swift
